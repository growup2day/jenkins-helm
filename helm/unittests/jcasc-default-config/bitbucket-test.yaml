templates:
  - config-jcasc.yaml
tests:
  - it: should include bitbucket config in default config scripts when configured
    set:
      JCasC:
        defaultConfig: true
        anzBitbucket:
          enabled: true
          manageHooks: true
          credentialsId: "anz-bitbucket-cred"
    asserts:
      - isNotNull:
          path: data.[jcasc-default-config-bitbucket.yaml]
      - matchSnapshot:
          path: data.[jcasc-default-config-bitbucket.yaml]
  - it: should ignore credential id when manage hooks disabled
    set:
      JCasC:
        defaultConfig: true
        anzBitbucket:
          enabled: true
          manageHooks: false
          credentialsId: "anz-bitbucket-cred"
    asserts:
      - isNotNull:
          path: data.[jcasc-default-config-bitbucket.yaml]
      - matchSnapshot:
          path: data.[jcasc-default-config-bitbucket.yaml]
  - it: should include blank bitbucket config in default config scripts when not configured
    set:
      JCasC:
        defaultConfig: true
    asserts:
      - isNotNull:
          path: data.[jcasc-default-config-bitbucket.yaml]
      - matchSnapshot:
          path: data.[jcasc-default-config-bitbucket.yaml]
  - it: should not include bitbucket config in default config scripts when there is custom config script for the section
    set:
      JCasC:
        defaultConfig: true
        configScripts:
          custom-override: |
            unclassified:
              bitbucketEndpointConfiguration:
                endpoints: []
    asserts:
      - isNull:
          path: data.[jcasc-default-config-bitbucket.yaml]
      - isNotNull:
          path: data.[custom-override.yaml]
