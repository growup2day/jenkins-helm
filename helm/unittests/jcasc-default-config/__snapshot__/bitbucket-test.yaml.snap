should ignore credential id when manage hooks disabled:
  1: |
    |-
      unclassified:
        # Configure System > Bitbucket Endpoints
        bitbucketEndpointConfiguration:
          endpoints:
            - bitbucketServerEndpoint:
                displayName: "ANZ_Bitbucket"
                serverUrl: "https://bitbucket.nz.service.anz"
                manageHooks: false
should include bitbucket config in default config scripts when configured:
  1: |
    |-
      unclassified:
        # Configure System > Bitbucket Endpoints
        bitbucketEndpointConfiguration:
          endpoints:
            - bitbucketServerEndpoint:
                displayName: "ANZ_Bitbucket"
                serverUrl: "https://bitbucket.nz.service.anz"
                manageHooks: true
                credentialsId: "anz-bitbucket-cred"
should include blank bitbucket config in default config scripts when not configured:
  1: |
    |-
      unclassified:
        # Configure System > Bitbucket Endpoints
        bitbucketEndpointConfiguration:
          endpoints: []
